listenbrainz/mbid_mapping_writer/mbid_mapper.py

```diff
@@ -0,0 +1,284 @@
+                return (hit, MATCH_TYPE_MED_QUALITY)
+
+            # Poor results so far, lets try detuning fields and try again
+            if ac_dist > self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE and ac_detuned:
+                ac_hit = ac_detuned
+                ac_detuned = ""
+                self.is_ac_detuned = True
+                self._log(Markup(f"""<b>no match</b>, ac_dist too high {ac_dist:.3f} continuing: detune ac"""))
+                continue
+
+            if r_dist > self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE and r_detuned:
+                r_hit = r_detuned
+                r_detuned = ""
+                self.is_r_detuned = True
+                self._log(Markup(f"""<b>no match</b>, r_dist too high {r_dist:.3f} continuing: detune r"""))
+                continue
+
+            self._log(Markup(f"""<b>no good match</b>, ac_dist {ac_dist:.3f}, r_dist {r_dist:.3f}, moving on"""))
+            return (None, MATCH_TYPE_NO_MATCH)
+
+    def lookup(self, artist_credit_name_p, recording_name_p):
+
+        query = artist_credit_name_p + " " + recording_name_p
+        if self.remove_stop_words:
+            cleaned_query = []
+            for word in query.split(" "):
+                if word not in ENGLISH_STOP_WORD_INDEX:
+                    cleaned_query.append(word)
+
+            query = " ".join(cleaned_query)
+
+        search_parameters = {
+            'q': query,
+            'query_by': "combined",
+            'prefix': 'no',
+            'num_typos': self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE
+        }
+
+        while True:
+            try:
+                hits = self.client.collections[COLLECTION_NAME].documents.search(
+                    search_parameters)
+                break
+            except requests.exceptions.ReadTimeout:
+                print("Got socket timeout, sleeping 5 seconds, trying again.")
+                sleep(5)
+
+        if len(hits["hits"]) == 0:
+            return None
+
+        return hits["hits"][0]
+
+    def search(self, artist_credit_name, recording_name):
+        """
+            Main query body: Prepare the search query terms and prepare
+            detuned query terms. Then attempt to find the given search terms
+            and if not found, sequentially try the detuned versions of the
+            query terms. Return a match dict (properly formatted for this
+            query) or None if not match.
+        """
+
+        artist_credit_name_p = prepare_query(artist_credit_name)
+        recording_name_p = prepare_query(recording_name)
+
+        ac_detuned = prepare_query(self.detune_query_string(artist_credit_name, True))
+        r_detuned = prepare_query(self.detune_query_string(recording_name, False))
+        self._log(f"ac_detuned: '{ac_detuned}' r_detuned: '{r_detuned}'")
+
+        self.is_ac_detuned = False
+        self.is_r_detuned = False
+
+        while True:
+            hit = self.lookup(artist_credit_name_p, recording_name_p)
```

Comment:
This part can be extracted to method like `evaluate_and_lookup_hit` and call that method for the three cases. 
1) original names
2) ac detuned r original
3) ac detuned r detuned

The intent is to eliminate the while True loop and many local flags. IMO, that makes the code more readable. But since you mostly work with this code, feel free to choose the version you prefer.